docs/content/singleton-actors.md

```diff
@@ -0,0 +1,79 @@
+# Singleton Actors
+
+Singleton Actors allow a single source of truth for both operations and data across a multi-server simulation. They are server-side authoritative [Actors](https://docs.unrealengine.com/en-us/Programming/UnrealArchitecture/Actors) that are restricted to one instantiation on SpatialOS. Singleton Actors allow a single source of truth for both operations and data across a multi-server simulation. There are two kinds of Singleton Actors:
+
+* **Public Singleton Actors** - Singleton Actors which are replicated to clients. `GameState` is a Public Singleton Actor.
+* **Private Singletons** - Singleton Actors which are not accessible to clients. `GameMode` is a Private Singleton Actor.
+
+You can define any class as a Singleton Actor. At the moment the Unreal GDK only supports Public Singleton Actors.
+
+Each server should instantiate their own local version of each Singleton Actor. For `GameMode` and `GameState`, Unreal does this automatically.
+
+Due to Unreal servers spawning their own instances of each Singleton Actor, proper replication and authority management of Singleton Actors becomes a bit tricky. To solve this issue, we have introduced the concept of a Global State Manager (GSM) to enable proper replication of Singleton Actors. The GSM solves the problem of replicating Singleton Actors by only allowing the server with [authority](https://docs.improbable.io/reference/13.1/shared/glossary#read-and-write-access-authority) over the GSM to execute the initial replication of these Actors. All other servers will then link their local Singleton Actors to their respective SpatialOS entity.
+
+## Setting up Singleton Actors
+
+To properly set up Singleton Actors for your project, you need to
+1. Register Singleton Actors with the DefaultEditorSpatialGDK.ini file
+1. Add the generated components to the UnrealWorker config
+
+### Registering Singleton Actors
+
+You need to specify Singleton Actors in `DefaultEditorSpatialGDK.ini` before creating the snapshot and running SpatialOS.
+
+Using the [Unreal GDK Starter Project](https://github.com/improbable/UnrealGDKStarterProject) as an example, to do this:
+1. Locate the `DefaultEditorSpatialGDK.ini` file in the Starter Project repository which you cloned during installation and setup - it's located at `Game/Config/DefaultEditorSpatialGDK.ini`.
+2. Open the file in your editor and add your Singleton Actors as shown in the file snippet below.
+
+### Example file snippet
+
+In the snippet below, your game is `ExampleGame` and the Singleton Actor class you want to add is `ExampleGameGameState`
+
+[SnapshotGenerator.SingletonActorClasses]
+ExampleGameGameState=
+
+[InteropCodeGen.ClassesToGenerate]
+;ClassName=include_path.h
+;Leave empty if no includes required.
+
+;GameState
```

Comment:
We have not been consistent with our ini file comments, but I believe the comment above a typebinding group should match the name of the typebinding though. So replace GameState with ExampleGameGameState.